### DISCUSSION TITLE
Sqitch requires annotation vs plan

### DISCUSSION THREAD

**Discussion by bever1337** [theory/sqitch#623] [Q&A] [answered]
Howdy, thanks for your time as a reader.

I'm new to sqitch and I am teaching myself the tooling with a personal project. In my example, I have created table 'Foo' which requires 'SchemA'. Later, I create table 'Bar'. Due to some arbitrary requirements, 'Foo' should have a foreign key pointing to 'Bar'. To make things complicated, because I have not deployed (or created a new version) I would prefer to simply edit the Foo sql scripts and sqitch plan instead of creating a new migration.

The actual question: When updating the 'requires' statements, what is the difference between the sqitch plan vs the annotation comments in the sql scripts? I've noticed errors originating from a bad sqitch plan, for example, which indicates sqitch plan is the source of truth.

**Comment by theory** [comment, marked as answer]
Great question. Yes, the plan is the source of truth. The comments in the file are for the convenience of the developer, as a reference. More than once I've failed to manually update them when I've changed the plan. :-/

**Reply by bever1337** [reply]
Awesome, thanks so much for your timeliness and help!
